ui/src/app/pages/analysis/notebook-list.tsx

```diff
@@ -36,23 +50,68 @@ export const NotebookList = withCurrentWorkspace()(class extends React.Component
     notebookList: FileDetail[],
     notebookNameList: string[],
     creating: boolean,
-    loading: boolean
+    loading: boolean,
+    delay: number,
+    cloneLoadingNotebookMsg: boolean;
   }> {
+  private interval: NodeJS.Timeout;
   constructor(props) {
     super(props);
-    this.state = {notebookList: [], notebookNameList: [], creating: false, loading: false};
+    this.state = {
+      notebookList: [],
+      notebookNameList: [],
+      creating: false,
+      loading: false,
+      delay: 3000,
+      cloneLoadingNotebookMsg: false
+    };
   }
 
   componentDidMount() {
     this.props.hideSpinner();
     profileApi().updatePageVisits({page: 'notebook'});
-    this.loadNotebooks();
+    this.confirmNotebookTransferIsDone();
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps, prevState) {
     if (this.workspaceChanged(prevProps)) {
       this.loadNotebooks();
     }
+    if (prevState.delay !== this.state.delay) {
+      clearInterval(this.interval);
+      this.interval = setInterval(this.tick, this.state.delay);
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  async confirmNotebookTransferIsDone() {
+    await workspacesApi()
+        .notebookTransferComplete(this.props.workspace.namespace, this.props.workspace.id).then((transferDone) => {
+          if (!transferDone) {
+            // Set the interval so that file transfer check is done after every 3 sec
+            this.interval = setInterval(this.tick, this.state.delay);
+            this.setState({loading: true, cloneLoadingNotebookMsg: true});
+          } else {
+            // Notebook transfer is done load the notebooks
+            this.loadNotebooks();
```

Comment:
should set `cloneLoadingNotebookMsg` to false, either here or inside loadNotebooks()

Reply:
Since by default the value cloneLoadingNotebookMsg is false i am not changing it here. 

In case of Tick or when the notebooks are not ready the first time, its been set to false at 109.